app/javascript/components/CardColumn/index.js

```diff
@@ -81,24 +78,42 @@ const CardColumn = ({kind, initCards}) => {
   }, []);
 
   const card = cards.find(it => it.id === popupShownId);
+  const checkTestId = id => {
+    return id.toString().startsWith('tmp-');
+  };
+
   return (
     <>
-      <CardColumnHeader kind={kind} />
+      <CardColumnHeader
+        kind={kind}
+        currentUser={currentUser}
+        onCardAdded={cardAdded => {
+          setCards(oldCards => [cardAdded, ...oldCards]);
+        }}
+        onGetIdNewCard={(cardMockid, cardId) => {
+          setCards(oldCards => {
+            oldCards[
+              oldCards.findIndex(it => it.id === cardMockid)
+            ].id = cardId;
+            return oldCards;
+          });
+        }}
```

Comment:
It's not bad to create additional var :) I think it makes it more clear
```js
onGetIdNewCard={(cardMockid, cardId) =>
  setCards(oldCards => {
    const index = oldCards.findIndex(it => it.id === cardMockid)
    oldCards[index].id = cardId;
    return oldCards;
  });
}
```

Also there is a good opportunity to try [useCallback](https://ru.reactjs.org/docs/hooks-reference.html#usecallback) hook.